examples/pytorch/dagnn/utils.py

```diff
@@ -0,0 +1,29 @@
+import numpy as np
+import random
+from torch.nn import functional as F
+import torch
+
+
+def evaluate(model, graph, feats, labels, idxs):
+    model.eval()
+    with torch.no_grad():
+        logits = model(graph, feats)
+        results = ()
+        for idx in idxs:
+            loss = F.cross_entropy(logits[idx], labels[idx])
+            acc = torch.sum(logits[idx].argmax(dim=1) == labels[idx]).item() / len(idx)
+            results += (loss, acc)
+    return results
+
+
+def generate_random_seeds(seed, nums):
+    random.seed(seed)
+    return [random.randint(1, 999999999) for _ in range(nums)]
+
+
+def set_random_state(seed):
+    random.seed(seed)
+    np.random.seed(seed)
+    torch.manual_seed(seed)
+    torch.cuda.manual_seed_all(seed)
```

Comment:
Is this valid if cuda is not available?

Reply:
I have added codes to check if cuda is not available.